scripts/build/builders/tizen.py

```diff
@@ -130,9 +131,55 @@ def __init__(self,
             raise Exception("TSAN sanitizer not supported by Tizen toolchain")
         if use_ubsan:
             self.extra_gn_options.append('is_ubsan=true')
+        self.use_coverage = use_coverage
+        if use_coverage:
+            self.extra_gn_options.append('use_coverage=true')
         if with_ui:
             self.extra_gn_options.append('chip_examples_enable_ui=true')
 
+    def generate(self):
+        super(TizenBuilder, self).generate()
+        if self.app == TizenApp.TESTS and self.use_coverage:
+            self.coverage_dir = os.path.join(self.output_dir, 'coverage')
+            self._Execute(['mkdir', '-p', self.coverage_dir], title="Create coverage output location")
+
+    def PreBuildCommand(self):
+        if self.app == TizenApp.TESTS and self.use_coverage:
+            cmd = ['ninja', '-C', self.output_dir]
+
+            if self.ninja_jobs is not None:
+                cmd.append('-j' + str(self.ninja_jobs))
+
+            cmd.append('Tizen')
+
+            self._Execute(cmd, title="Build-only")
+
+            gcov = os.path.join(os.environ['TIZEN_SDK_TOOLCHAIN'], 'bin/arm-linux-gnueabi-gcov')
+
+            self._Execute(['lcov', '--gcov-tool', gcov, '--initial', '--ignore-errors', 'unused', '--capture', '--directory', os.path.join(self.output_dir, 'obj'),
+                           '--exclude', '**/src/controller/*',
```

Comment:
this should probably be in some separate list as it is repeated and very independent from the other arguments.

Reply:
It looks like we want some shared gcov arguments and then only have the diffs inside build and post build.

Reply:
Is it okay like this?